### ISSUE TITLE
Cannot read property 'getActiveChildNavs' of undefined

### ISSUE THREAD

**Issue by traed** [zyra/ionic2-super-tabs#308]
Getting the following error when switching between pages containing super tabs (using nav.setRoot from the sidebar). Going (Page with supertabs) -> (Page without supertabs) -> (Other page with super tabs) works but (Page with supertabs) -> (Other page with super tabs) does not. The pages are all instances of the same ionic page template but with different data for the super tabs if that matters. Lazy loading is on.

> Error: Uncaught (in promise): TypeError: Cannot read property 'getActiveChildNavs' of undefined
> TypeError: Cannot read property 'getActiveChildNavs' of undefined
>     at findTopNavs (http://localhost:8100/build/vendor.js:27223:25)
>     at http://localhost:8100/build/vendor.js:27229:27
>     at Array.forEach (<anonymous>)
>     at findTopNavs (http://localhost:8100/build/vendor.js:27228:19)
>     at http://localhost:8100/build/vendor.js:27229:27
>     at Array.forEach (<anonymous>)
>     at findTopNavs (http://localhost:8100/build/vendor.js:27228:19)
>     at http://localhost:8100/build/vendor.js:27047:32
>     at Map.forEach (<anonymous>)
>     at App.getActiveNavContainers (http://localhost:8100/build/vendor.js:27046:24)
>     at c (http://localhost:8100/build/polyfills.js:3:19752)
>     at new t (http://localhost:8100/build/polyfills.js:3:21532)
>     at __awaiter (http://localhost:8100/build/vendor.js:65399:12)
>     at SuperTabs.ngAfterViewInit (http://localhost:8100/build/vendor.js:65618:16)
>     at callProviderLifecycles (http://localhost:8100/build/vendor.js:13122:18)
>     at callElementProvidersLifecycles (http://localhost:8100/build/vendor.js:13089:13)
>     at callLifecycleHooksChildrenFirst (http://localhost:8100/build/vendor.js:13072:29)
>     at checkAndUpdateView (http://localhost:8100/build/vendor.js:14227:5)
>     at callViewAction (http://localhost:8100/build/vendor.js:14569:21)
>     at execEmbeddedViewsAction (http://localhost:8100/build/vendor.js:14527:17)

Super tabs markup:
```
<super-tabs indicatorColor="secondary" scrollTabs="true" id="super-tabs" [config]="{ sideMenu: 'left' }" no-shadow *ngIf="categories.length > 0; else noContentBlock">
	<super-tab [root]="'ProductCategoryPage'" title="{{category.title}}" [rootParams]="category.params" *ngFor="let category of categories"></super-tab>
</super-tabs>
```

Version info:
Ionic Framework: 3.9.2
Ionic App Scripts: 3.1.9
Angular Core: 5.2.10
Angular Compiler CLI: 5.2.10
Node: 10.0.0
OS Platform: macOS Sierra
Navigator Platform: MacIntel
User Agent: Mozilla/5.0 (iPhone; CPU iPhone OS 11_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Mobile/15A372 Safari/604.1

**Comment by vibinflogesoft** [comment]
Hi@traed,
Iam having the same issue.Did you get any solution for this?

**Comment by traed** [comment]
Haven't really gotten a solution. The issue occurs when super-tabs only has one tab so my workaround is just adding an ngIf to the super-tabs element to prevent super-tabs from running if there is only one element like so:
```
<super-tabs *ngIf="someArray.length > 1; else singleCategory">
  <super-tab *ngFor="let someItem of someArray">{{Some content}}</super-tab>
</super-tabs>
<ng-template #singleCategory>{{Same content as above}}</ng-template>
```
Not as DRY as I would like but it works.

**Comment by ChrizC** [comment]
Hi - I've submitted a fix for this in PR #323 - if you could show support for that PR we can get it fixed properly

**Comment by Thomas0921** [comment]
I came across this problem too, I solved this by using the version "ionic2-super-tabs": "^4.1.4", and I had made a mistake in my ngFor loop in super tab tag (since my array of object is a nested array). hope that helps :)
